### PR TITLE
feat: [#495] Centralize Error Messages in Framework (PR#1)

### PR DESCRIPTION
**Pull request by kkumar-gcc**
### Example 
```go
ErrDriverNotSupported := errors.New("driver [%s] not supported", "session")

func Driver(driver string) (any, error) {
       // withLocation default is true
       return nil, ErrDriverNotSupported.Args(driver).WithLocation(true)
}

_, err := Driver("file")
print(err)
// [session] Driver [file] not supported
```

RelatedTo https://github.com/goravel/goravel/issues/495

## Summary by CodeRabbit

- **New Features**
	- Introduced a custom error handling mechanism that provides detailed error messages with optional location context.
	- Added an interface for enhanced error management, allowing users to set arguments and control the inclusion of location in error messages.

These changes improve the clarity and context of error reporting within the application.

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

**Comment by kkumar-gcc** [comment]
@coderabbitai summary

File: errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+
+	errorscontracts "github.com/goravel/framework/contracts/errors"
```
Comment (hwbrzzl) [inline errors/errors.go:7]:
```suggestion
	contractserrors "github.com/goravel/framework/contracts/errors"
```

**Review by hwbrzzl — COMMENTED** [review]
How to use it? Add an example?

**Comment by kkumar-gcc** [comment]
Added an example

File: contracts/errors/errors.go
Comment (hwbrzzl) [inline contracts/errors/errors.go:1]: It seems unnecessary.

File: errors/errors.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	if e.withLocation && e.location != "" {
+		formattedText = fmt.Sprintf("[%s] %s", e.location, formattedText)
+	}
+
+	return formattedText
+}
+
+func (e *errorString) Location(location string) contractserrors.Error {
```
Comment (hwbrzzl) [inline errors/errors.go:50]: `Location` vs `Module`?

File: errors/errors.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	e.location = location
+	return e
+}
+
+func (e *errorString) WithLocation(flag bool) contractserrors.Error {
```
Comment (hwbrzzl) [inline errors/errors.go:55]: It seems unnecessary.

File: errors/errors.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	return errors.Is(err, target)
+}
+
+func As(err error, target any) bool {
+	return errors.As(err, &target)
+}
+
+func Unwrap(err error) error {
+	return errors.Unwrap(err)
+}
```
Comment (hwbrzzl) [inline errors/errors.go:70]: We can add them when needed.

File: errors/errors_list.go
Comment (hwbrzzl) [inline errors/errors_list.go:1]: Rename the file name to `list.go`

File: session/service_provider.go [hunk 5]
```diff
@@ -20,12 +21,12 @@ func (receiver *ServiceProvider) Register(app foundation.Application) {
 	app.Singleton(Binding, func(app foundation.Application) (any, error) {
 		c := app.MakeConfig()
 		if c == nil {
-			return nil, ErrConfigFacadeNotSet
+			return nil, errors.ErrConfigFacadeNotSet
```
Comment (hwbrzzl) [inline session/service_provider.go:24]: Should set location, define a constant: errors.ModuleSession?

File: session/service_provider.go [hunk 6]
```diff
@@ -20,12 +21,12 @@ func (receiver *ServiceProvider) Register(app foundation.Application) {
...
 		}
 
 		j := app.GetJson()
 		if j == nil {
-			return nil, ErrJSONNotSet
+			return nil, errors.ErrJSONParserNotSet
```
Comment (hwbrzzl) [inline session/service_provider.go:29]: Ditto

File: contracts/errors/errors.go
Comment (kkumar-gcc) [reply contracts/errors/errors.go:1]: Hmm

File: errors/errors.go (hunk 4 above)
Comment (kkumar-gcc) [reply errors/errors.go:70]: `Is` is being used at some places

**Review by hwbrzzl — APPROVED** [review]
Great 👍

**Pull request merged**